xpk.py

```diff
@@ -1127,6 +1142,120 @@ def zone_to_region(zone) -> str:
   return zone_terms[0] + '-' + zone_terms[1]
 
 
+def get_service_account_name(args) -> str:
+  """Get the name for the service account.
+  Args:
+    args: user provided arguments.
+
+  Returns:
+    the name of the service account.
+  """
+  return f'{args.project}-{_XPK_SERVICE_ACCOUNT}@{args.project}.iam.gserviceaccount.com'
+
+def check_if_service_account_exists(args) -> bool:
+  """Check if a service account with the given name exists in the project.
+
+  Args:
+    args: user provided arguments for running the command.
+
+  Returns:
+    True if service account exist, False otherwise.
+  """
+  service_account_name = get_service_account_name(args)
+  command = (
+    f'gcloud iam service-accounts describe {service_account_name}'
+  )
+  return_code = run_command_with_updates(command, 'Service Account Describe', args, verbose=False)
+  if return_code != 0:
+    xpk_print(f'Service Account Describe did not find the service account {service_account_name}.')
+    return False
+  return True
+
+def create_service_account(args) -> int:
+  """Creates a service account in the project.
+
+  Args:
+    args: user provided arguments for running the command.
+
+  Returns:
+    0 if successful and 1 otherwise.
+  """
+  command = (
+    f'gcloud iam service-accounts create {args.project}-{_XPK_SERVICE_ACCOUNT} \
+    --description="Service Account for XPK" \
+    --display-name="{args.project}-{_XPK_SERVICE_ACCOUNT}"'
+  )
+  return_code = run_command_with_updates(command, 'Service Account Create', args)
+  if return_code != 0:
+    xpk_print(f'Service Account Create request returned ERROR {return_code}')
+    xpk_print('Make sure you have Service Account Admin Role attached to your user.')
+    return 1
+  return 0
+
+def get_existing_roles_in_service_account(args) -> set:
+  """
+  Args:
+    args: user provided arguments for running the command.
+
+  Returns:
+    set of IAM roles already attached to the service account.
+  """
+  roles = []
+  service_account_name = get_service_account_name(args)
+  command = (
+    f'gcloud projects get-iam-policy {args.project}'
+    f' --filter="bindings.members:{service_account_name}"'
+    ' --flatten="bindings[].members" --format="table(bindings.role)"'
+  )
+  return_code, return_value = run_command_for_value(command, 'Get IAM Roles For Service Account', args)
+  if return_code != 0:
+    xpk_print(f'Get IAM Roles For Service Account request returned ERROR {return_code}')
```

Comment:
If return_code is !=0, we hit this print statement but the function will return an empty list of existing roles and continue running. LMK if that is the expected behavior you want or do you want xpk to exit early?

Reply:
Yeah, this is the expected behavior. When this method returns an empty set of roles, then `add_roles_to_service_account()` will add the list of predefined roles to the service account. However, if by chance the role is already attached to the SA, the method won't do anything. So, we can keep the xpk running here.

Reply:
Is it possible `return_code, return_value = run_command_for_value(command, 'Get IAM Roles For Service Account', args)` will fail for other reasons besides returning an empty set of roles that we would want to abort for? 

If not, this sounds good to me!

Reply:
Yes, it may fail if there is no `resourcemanager.projects.getIamPolicy` permission attached to user. However, I think failing xpk here won't make sense. This is because we are maily interested in attaching the appropriate permissions to the SA. In case, user has `resourcemanager.projects.setIamPolicy` but not `resourcemanager.projects.getIamPolicy`, then they should be able to set the permissions to SA, which is all we want for the cluster to run.